lib/listener/mac/ITunesListener.cpp

```diff
@@ -45,26 +45,46 @@ struct ITunesConnection : PlayerConnection
 
 
 ITunesListener::ITunesListener( QObject* parent )
-    :QObject( parent ), m_connection( 0 )
+    : QObject( parent )
+    , m_previousPid()
+    , m_playerAppId( getPlayerAppId() )
+    , m_connection( 0 ) 
 {
+    qDebug() << "Detected player app ID: " << m_playerAppId;
     qRegisterMetaType<Track>("Track");
 
-    m_currentTrackScript = AppleScript("tell application \"iTunes\" to tell current track\n"
-                                          "try\n"
-                                              "set L to location\n"
-                                              "set L to POSIX path of L\n"
-                                          "on error\n"
-                                              "set L to \"\"\n"
-                                          "end try\n"
-                                          "return artist & \"\n\" & album artist & \"\n\" & album & \"\n\" & name & \"\n\" & (duration as integer) & \"\n\" & L & \"\n\" & persistent ID & \"\n\" & podcast & \"\n\" & video kind\n"
-                                      "end tell\n" );
-
-    CFNotificationCenterAddObserver( CFNotificationCenterGetDistributedCenter(),
-                                    this,
-                                    callback,
-                                    CFSTR( "com.apple.iTunes.playerInfo" ),
-                                    NULL,
-                                    CFNotificationSuspensionBehaviorDeliverImmediately );
+    QString mediaTypeFields = m_playerAppId == "com.apple.Music"
+        ? "\"false\" & \"\n\" & \"none\"\n"
```

Comment:
So "false" and "none" are valid values for podcast and video kind?

Reply:
They are parsed by us later on here - https://github.com/lastfm/lastfm-desktop/blob/master/lib/listener/mac/ITunesListener.cpp#L200#L202